packages/@datapull/kinesis-stream-destination/util/batch-splitter.js

```diff
@@ -0,0 +1,48 @@
+module.exports = function (records, callback) {
+  var batches = [];
+
+  var currentBatch = [];
+  var currentBatchMessagesLength = 0;
+
+  records.forEach(function (record) {
+    if (!record.Data) {
+      console.warn("WARNING, message should be wrapped in 'Data' field");
+      return;
+    }
+
+    if (!record.PartitionKey) {
+      console.warn("WARNING, message should have a 'PartitionKey' field");
+      return;
+    }
+
+    if (record.Data.length >= 1000 * 1000) {
+      console.error("WARNING, the message size exceeds 1000kb, cannot be sent to Kinesis", record.Data.length, record._meta);
+      return;
+    }
+
+    currentBatch.push(record);
+    currentBatchMessagesLength += record.Data.length + record.PartitionKey.length;
+
+    // no more than 500 records in every batch
```

Comment:
Is there any reasoning behind the limit of 500?

Reply:
that's a limit on the Amazon Kinesis PutRecords operation